spec/filters/grok_spec.rb

```diff
@@ -702,4 +702,66 @@
     end
   end
 
+  describe  "grok with nil coerced value" do
+    config <<-CONFIG
+      filter {
+        grok {
+          match => { "message" => "test (N/A|%{BASE10NUM:duration:float}ms)" }
+        }
+      }
+    CONFIG
+
+    sample "test 28.4ms" do
+      insist { subject["duration"] } == 28.4
+      insist { subject["tags"] }.nil?
+    end
+
+    sample "test N/A" do
+      insist { insist { subject.to_hash }.include?("duration") }.fails
```

Comment:
`.fails` is for exceptions, basically asserting that any (I think?) exception is raised.

insist objects support predicate methods, so you can do this if you just want to check if a thing includes a thing:
```
insist { subject.to_hash }.include?("duration")
```

Some examples from insist's test suite on predicates:
* https://github.com/jordansissel/ruby-insist/blob/master/spec/insist/predicates_spec.rb
*